charts/flagsmith/templates/_helpers.tpl

```diff
@@ -308,3 +308,12 @@ args:
   - 'URL: %{url_effective}\nHTTP status code: %{http_code}\nBytes downloaded: %{size_download}\nTime taken: %{time_total}s\n'
   - {{ .url | squote }}
 {{- end }}
+
+{{/*
+Replicas
+*/}}
+{{- define "flagsmith.replicaCount" -}}
```

Comment:
A couple of questions that I'd like to understand here: 

 1. When does this helper get run? 
 2. The `values.yml` file defines `replicacount` (e.g. [here](https://github.com/Flagsmith/flagsmith-charts/blob/main/charts/flagsmith/values.yaml#L16)) without the capital C. is this an issue? As I understand it, yaml _is_ case sensitive so wouldn't this always fail if people are taking their values from the template we've provided? 

Also, to confirm my understanding, this code essentially defines the following logic: 

 1. Define an attribute `flagsmith.replicaCount`
 2. If that attribute already existed (presumably because it had already been set by values.yaml), then we define an attribute of `replicas` which can be included elsewhere in the templates
 3. If that attribute doesn't already exist, then we do nothing, and hence when it's included in other templates, nothing is _actually_ included.

Reply:
Think of this helper as a function like:
```
def flagsmith_replicaCount(x):
    if x is not None:
        print("replicas:", x)
```
Then elsewhere, you see it being "called" using `include`, where we pass things like `.Values.api.replicacount` to it as the argument. Yes it's slightly weird that the casing doesn't match (I'd argue it should probably all be `replicaCount`, but changing that would break backwards compatibility), but the behaviour is what we want.

Reply:
Ok, thanks for explaining.